Packs/BitSight/Integrations/BitSightEventCollector/BitSightEventCollector.py

```diff
@@ -0,0 +1,349 @@
+
+        Returns:
+            dict[str, Any]: The response payload from Bitsight's `/v1/companies` API, containing
+            information about the authorized company (e.g., `myCompany`) and related metadata.
+        """
+        return self._http_request(method="GET", url_suffix="v1/companies")
+
+    def get_company_findings(
+        self,
+        guid: str,
+        first_seen_gte: str,
+        last_seen_lte: str,
+        limit: int,
+        offset: int,
+    ) -> dict[str, Any]:
+        """Get company findings for given guid and time range.
+
+        Args:
+            guid (str): guid of the company whose findings need to be retrieved
+            first_seen_gte (str): first seen date (YYYY-MM-DD) of the findings
+            last_seen_lte (str): last seen date (YYYY-MM-DD) of the findings
+            limit (int): maximum number of findings to return
+            offset (int): offset to begin returning findings from
+
+        Returns:
+            dict[str, Any]: findings response
+        """
+        params = {
+            "first_seen_gte": first_seen_gte,
+            "last_seen_lte": last_seen_lte,
+            "unsampled": "true",
+            "expand": "attributed_companies",
+            "limit": limit,
+            "offset": offset,
+        }
+        return self._http_request(method="GET", url_suffix=f"v1/companies/{encode_string_results(guid)}/findings", params=params)
+
+
+def to_bitsight_date(ts: int) -> str:
+    """Convert a UNIX timestamp (seconds) to Bitsight date format (YYYY-MM-DD).
+
+    Args:
+        ts (int): UNIX timestamp in seconds.
+
+    Returns:
+        str: Date string formatted as YYYY-MM-DD (UTC).
+    """
+    return datetime.fromtimestamp(ts, tz=UTC).strftime(BITSIGHT_DATE_FORMAT)
+
+
+def findings_to_events(findings: list[dict[str, Any]]) -> tuple[list[dict[str, Any]], list[str]]:
+    """Transform Bitsight findings into XSIAM event objects.
+
+    - Preserves original finding fields.
+    - Sets the XSIAM `_time` field using the finding's `first_seen` (YYYY-MM-DD) when available.
+    - Collects findings without dates for later error handling.
+
+    Args:
+        findings (list[dict[str, Any]]): A list of Bitsight findings as returned from the API.
+
+    Returns:
+        tuple[list[dict[str, Any]], list[str]]: A tuple containing:
+            - A list of event dictionaries suitable for pushing to XSIAM
+            - A list of finding IDs that lack date fields (for error handling after offset update)
+    """
+    events: list[dict[str, Any]] = []
+    missing_date_findings: list[str] = []
+
+    for f in findings:
+        event = dict(f)
+        # Set XSIAM time field to first_seen
+        first_seen = f.get("first_seen") or f.get("firstSeen")
+        if first_seen:
+            # Bitsight returns YYYY-MM-DD for first_seen
+            try:
+                dt = datetime.strptime(first_seen, BITSIGHT_DATE_FORMAT)
```

Comment:
please avoid naming vars with no context to their name - e.g. `dt` can be any time, or even a dt query. `first_seen_dt` will be better